### PR TITLE
Fix #17, Fix #19: Beef up the tests, as well as use provided config

### LINKED ISSUE #17: improve code coverage to 80%
Code coverage in the pipeline should be bumped to over 80% to satisfy pipeline-test-node (best practices) defaults
https://kenzan.ci.cloudbees.com/job/DEV-pipeline-validate-css/

### LINKED ISSUE #19: Pipeline is not utilizing the included .csslintrc file
When attempting to validate CSS, I discovered that I had to include a root-level `.csslintrc` file, when it's expected that the included file would be respected.

### PR DESCRIPTION
**Pull request by paul-barry-kenzan**
**Associated Issues**
#17 
#19 

**Changes Made**
  - Ensure that the provided options, whether a JSON object or file path, was used by the linter
  - Fallback to the default config when options are NOT provided
  - Abstract the custom formatter to a standalone, testable module
  - Increase test coverage to 100% for both the pipeline, and formatter

**Reviewers**
@thescientist13 
@bsawyer 
@cxoltero

### CODE REVIEW

**Comment by thescientist13** [comment]
Can one of the admins verify this patch?

**Comment by thescientist13** [comment]
ok to test

**Comment by thescientist13** [comment]
@paul-barry-kenzan 
Looks like you forgot to commit a dependency to _package.json_

**Comment by paul-barry-kenzan** [comment]
ok to test ?

File: src/formatter.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+var gulpUtil = require('gulp-util');
+var handyman = require('pipeline-handyman');
+
+module.exports = function cssLintFormatter (file) {
+  var color = gulpUtil.colors;
+
+  file.messages.forEach(function (result) {
+    var call = ' line ' + result.line + '   col ' + result.col + '   =>   ' + result.message;
```
Comment (thescientist13) [inline src/formatter.js]: `call`?  is there a context to this variable name that I am unfamiliar with?

File: package.json [hunk 2]
```diff
@@ -41,22 +41,23 @@
...
-    "gulp-filter": "4.0.0",
     "gulp-util": "3.0.8",
     "lazypipe": "1.0.1",
     "pipeline-handyman": "1.0.0"
   },
   "devDependencies": {
     "chai": "3.3.0",
+    "dirty-chai": "1.2.2",
+    "gulp": "3.9.1",
     "gulp-util": "3.0.6",
```
Comment (thescientist13) [inline package.json]: `gulp-util` is defined twice it seems

File: test/index.spec.js [hunk 3]
```diff
@@ -2,58 +2,180 @@
...
+      validateCSSPipeline.validateCSS();
+
+      expect(spy).to.have.been.calledWith(sinon.match.func);
+
+      cssLint.formatter.restore();
+    });
+
+    describe('validateCSS default usage', function () {
+
+      xit('should utilize cssLint with no options', function () {
```
Comment (thescientist13) [inline test/index.spec.js]: `xit` here

File: test/index.spec.js [hunk 4]
```diff
@@ -2,58 +2,180 @@
...
+        var spy = sinon.spy(handyman, 'mergeConfig');
+
+        validateCSSPipeline.validateCSS(customConfig);
+
+        expect(spy).to.have.been.calledWith(sinon.match.object, customConfig);
+
+        handyman.mergeConfig.restore();
+      });
+
+      xit('should utilize cssLint with the provided options object', function () {
```
Comment (thescientist13) [inline test/index.spec.js]: `xit` here.  intentional?

File: test/index.spec.js (hunk 3 above)
Comment (paul-barry-kenzan) [inline test/index.spec.js]: Yes. Reasoning in the comment below.

Comment (thescientist13) [inline test/index.spec.js]: can you make an issue for that?  we can reference this ticket https://github.com/kenzanlabs/keystone/issues/34

but I don't have any idea when native `imports` will be added to NodeJS.  The pipeline or  consumer would need to use babel or the harmony flag, right?

Comment (paul-barry-kenzan) [inline test/index.spec.js]: @thescientist13 
#27 and #28 have been created to track the update to ES6 and the enabling of those 2 tests.

File: src/index.js [hunk 5]
```diff
@@ -2,43 +2,51 @@
...
+  defaultConfig = fs.readFileSync(path.join(process.cwd(), '.csslintrc'), 'utf8');
+  defaultConfig = JSON.parse(defaultConfig.toString());
+
+  if (typeof options === 'object') {
+    config = handyman.mergeConfig(defaultConfig, options);
 
-  file.messages.forEach(function(result) {
-    var call = ' line ' + result.line + '   col ' + result.col + '   =>   ' + result.message;
+  } else if (typeof options === 'string') {
+    providedConfig = fs.readFileSync(path.join(process.cwd(), options), 'utf8');
```
Comment (thescientist13) [inline src/index.js]: when a `string` is passed, we presume the user is providing a path to their own custom _.csslintrc_ right?  If so, how come the logic to read _.csslintrc_ is the same for the default and custom cases?

If a user is providing their own custom config, and they pass say _./.csslintrc_ and we read it from `process.cwd()` wouldn't that effectively be the same as the default lookup logic?

Compared to what happens in _pipeline-validate-js_
https://github.com/kenzanlabs/pipeline-validate-js/blob/master/src/index.js#L43

Comment (paul-barry-kenzan) [inline src/index.js]: Nice catch, @thescientist13 . Was attempting to safe guard against a bad file path, but if the path is provided, it should be used as is. Will update this.

File: src/index.js [hunk 6]
```diff
@@ -2,43 +2,58 @@
...
+  var config;
+  var defaultConfig;
+  var providedConfig;
+
+  defaultConfig = fs.readFileSync(path.join(process.cwd(), '.csslintrc'), 'utf8');
+  defaultConfig = JSON.parse(defaultConfig.toString());
+
+  if (typeof options === 'object') {
+    config = handyman.mergeConfig(defaultConfig, options);
+
```
Comment (cxoltero) [inline src/index.js]: In ES6, there is a new way to combine objects, see [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).
syntax:
``` javascript
config = Object.assign(defaultConfig, providedConfig);
```

File: test/index.spec.js (hunk 3 above)
Comment (thescientist13) [inline test/index.spec.js]: let's add comments with a `TODO` for these two tests here referencing issue #28 and then I think we can close the loop on these `xit`s

**Pull request merged**